gateway/ln-gateway/src/lib.rs

```diff
@@ -227,16 +223,93 @@ impl Gateway {
                     "Gateway configured to connect to LND LnRpcClient at \n address: {:?},\n tls cert path: {:?},\n macaroon path: {} ",
                     lnd_rpc_addr, lnd_tls_cert, lnd_macaroon
                 );
-                Arc::new(RwLock::new(
-                    GatewayLndClient::new(lnd_rpc_addr, lnd_tls_cert, lnd_macaroon, task_group)
-                        .await?,
-                ))
+                Arc::new(GatewayLndClient::new(lnd_rpc_addr, lnd_tls_cert, lnd_macaroon).await?)
             }
         };
 
         Ok(lnrpc)
     }
 
+    pub async fn route_htlcs(&mut self, task_group: &mut TaskGroup) -> Result<()> {
+        // Create a stream used to communicate with the Lightning implementation
+        let (sender, ln_receiver) = mpsc::channel::<InterceptHtlcResponse>(100);
+
+        let mut lnrpc: Box<dyn ILnRpcClient> = match &self.lightning_mode {
+            Some(LightningMode::Cln { cln_extension_addr }) => {
```

Comment:
- For CLN would we have to make `subscribe_mint_htlcs` rpc before we open the stream with `route_htlcs` rpc? Maybe the CLN extension as currently implemented would just filter out all htlcs without forwarding any to gatewayd

 - If above is true, it's very strange that we need these RPCs to be called in a certain order, and that `subscribe_mint_htlcs` affects behavior of `route_htlcs`.
 
 - Second, we'd need a way of reversing a filter registered by `subscribe_mint_htlcs`

Reply:
Offline call with @okjodom. Going to remove the functionality for subscribing to HTLCs for now. This PR doesn't support the multiple `gatewayd` case and implementing that feature with the new client architecture likely requires more careful/scoped implementation in a separate PR.

Reply:
More generally, the paired behavior of `subscribe_mint_htlcs` and `route_htlcs` makes `route_htlcs` rpc unpredictable, and `subscribe_mint_htlcs` does not set clear API expectations to whoever is implementing the rpc.

After my discussion with @m1sterc001guy today, we resolved to remove `subscribe_mint_htlcs` rpc and consider re-introducing filtering on the edge at a later point. See #2626